### ISSUE TITLE
Unnecessary warning when calling Admin.deleteTopicRecords with offset -1

### ISSUE THREAD

**Issue by KillWolfVlad** [tulios/kafkajs#987]
**Describe the bug**

Admin Client log unnecessary warning when calling deleteTopicRecords with offset -1. Additional conditions may be missing.

https://github.com/tulios/kafkajs/blob/ba34eef11abdbeb69598947358fe220efb38d514/src/admin/index.js#L1045-L1054

**To Reproduce**

Simple code to reproduce issue:

```js
const TOPIC = "some_topic";

const {Kafka} = require("kafkajs");

const kafka = new Kafka({
  brokers: ["localhost:9092"],
});

const admin = kafka.admin();

(async () => {
  await admin.connect();

  const topicOffsets = await admin.fetchTopicOffsets(TOPIC);

  await admin.deleteTopicRecords({
    topic: TOPIC,
    partitions: topicOffsets.map((x) => ({
      partition: x.partition,
      offset: "-1",
    })),
  });

  await admin.disconnect();
})().catch((error) => {
  console.error(error);
  process.exit(1);
});
```

Stdout:

```json
{
  "level": "WARN",
  "timestamp": "1970-01-01T00:00:00.000Z",
  "logger": "kafkajs",
  "message": "[Admin] The requested offset is before the earliest offset maintained on the partition - no records will be deleted from this partition",
  "topic": "some_topic",
  "partition": 0,
  "offset": "-1"
}
```

**Expected behavior**

Admin Client must not log unnecessary warning.

**Comment by zacken** [comment]
Submitted pull request #1265 to address this issue

**Comment by Nevon** [comment]
Closed by #1265
